src/uct/ib/base/ib_md.c

```diff
@@ -2058,6 +2059,7 @@ static ucs_status_t uct_ib_verbs_md_open(struct ibv_device *ibv_device,
         goto err_md_free;
     }
 
+    md->reg_nonblock_mem_types = 0;
```

Comment:
maybe move it to uct_ib_md_open_common, along with the initialization of  md->reg_mem_types ?

Reply:
If I understand the logic correctly, then moving it to `uct_ib_md_open_common` will overwrite value that can be set during `uct_ib_mlx5_devx_check_odp`.

Reply:
ok
but probably we don't need it anyway since md is allocated by calloc

Reply:
explicit initialization is a good "code documentation". BTW, do we really need to call `uct_ib_md_open_common` so late? Would it be possible to call it earlier, so that it can initialize common stuff?

Reply:
IMO we can call it earlier from mlx5 md init code (devx and mlx5dv) but I would not do it in this PR since it may cause unexpected issues, and needs more testing.
we already rely on md zero initialization for some other fields, AFAIR we discussed it with @Artemy-Mellanox

Reply:
IMO we should keep relying on calloc for now and maybe refactor uct_ib_md_open_common to be called earlier. for example seems it does not rely on the initialization of ODP flags.

Reply:
ok, let's remove initialization in this PR and (maybe) refactor md open later in another PR